src/twitter/Timeline.js

```diff
@@ -240,50 +282,59 @@ class Timeline {
       this._newtags.delete(req.user_id);
     }
 
-    this.twitter.storage.get_since("@"+tline.twitter_id).then((since) => {
-      log.silly("Polling %s, since value: %s", "@"+tline.twitter_id, since);
-      if (since) {
-        req.since_id = since;
-      }
-      return this.twitter.client_factory.get_client();
-    }).then((client)=>{
-      return client.getAsync('statuses/user_timeline', req).catch((error) =>{
-        log.error("Timeline", "_process_timeline: GET /statuses/user_timeline returned: %s", error.code);
-      });
-    }).then((feed) => {
-      if (feed.length === 0) {
-        return;
-      }
-      else if(feed.length === TIMELINE_TWEET_FETCH_COUNT) {
-        log.info("Timeline poll request hit count limit. Request likely incomplete.");
+    const since = yield this.twitter.storage.get_since("@"+tline.twitter_id);
+    if (since) {
+      req.since_id = since;
+    }
+    let feed;
+    try {
+      feed = yield client.getAsync('statuses/user_timeline', req);
+    }
+    catch (error) {
+      log.error("Timeline", "_process_timeline: GET /statuses/user_timeline returned: %s", error.code);
+      return;
+    }
+    if (feed.length === 0) {
+      return;
+    }
+    else if(feed.length === TIMELINE_TWEET_FETCH_COUNT) {
+      log.info("Timeline", "Poll request hit count limit. Request likely incomplete.");
```

Comment:
Might be worth mention which timeline. I would suggest putting `tline.twitter_id` in logs for `process_timeline` and `process_hashtags`